### PR TITLE
CATROID-213 Add new user defined brick

### PR DESCRIPTION
**Pull request by DollyVolley**
Add new Brick Layout
Add new User Brick Fragment

https://jira.catrob.at/browse/CATROID-213

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/UserBrickListFragment.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+		return view;
+	}
+
+	@Override
+	public void onClick(View v) {
+		AddUserBrickFragment addUserBrickFragment = AddUserBrickFragment.newInstance();
+		String tag = AddUserBrickFragment.ADD_USER_BRICK_FRAGMENT_TAG;
+
+		getFragmentManager().beginTransaction()
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/UserBrickListFragment.java:79]: Please make a null-ptr check.

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/UserBrickListFragment.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	public static UserBrickListFragment newInstance() {
+		UserBrickListFragment fragment = new UserBrickListFragment();
+
+		return fragment;
+	}
+
+	@Override
+	public void onDestroyView() {
+		super.onDestroyView();
+		((AppCompatActivity) getActivity())
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/UserBrickListFragment.java:57]: Please make a null-ptr check.

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/UserBrickListFragment.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	@Override
+	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+		View view = inflater.inflate(R.layout.fragment_user_brick_list, container, false);
+		setHasOptionsMenu(true);
+		addUserBrickButton = view.findViewById(R.id.add_user_brick);
+		addUserBrickButton.setOnClickListener(this);
+
+		((AppCompatActivity) getActivity())
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/UserBrickListFragment.java:68]: Please make a null-ptr check.

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserBrickFragment.java [hunk 4]
```diff
@@ -45,9 +49,30 @@ public static AddUserBrickFragment newInstance() {
...
+		((AppCompatActivity) getActivity())
+				.getSupportActionBar().setTitle(R.string.category_user_bricks);
+	}
+
+	private void handleAddLabel() {
+		int i = 1;
+	}
+
+	private void handleAddInput() {
+		int i = 1;
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserBrickFragment.java:76]: Here too, please.

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserBrickFragment.java [hunk 5]
```diff
@@ -45,9 +49,30 @@ public static AddUserBrickFragment newInstance() {
...
-				.getSupportActionBar().setTitle(R.string.category_user_bricks);
+				.getSupportActionBar().setTitle(R.string.brick_add_new_user_brick);
+
 		return view;
 	}
+
+	@Override
+	public void onDestroyView() {
+		super.onDestroyView();
+		((AppCompatActivity) getActivity())
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserBrickFragment.java:67]: Null-ptr check, please.

File: catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserBrickFragment.java [hunk 6]
```diff
@@ -45,9 +49,30 @@ public static AddUserBrickFragment newInstance() {
...
-		View view = inflater.inflate(R.layout.fragment_brick_add_userbrick, container, false);
+		View view = inflater.inflate(R.layout.fragment_add_new_user_brick, container, false);
+		addLabel = view.findViewById(R.id.add_label);
+		addInput = view.findViewById(R.id.add_input);
+		addLabel.setOnClickListener(v -> handleAddLabel());
+		addInput.setOnClickListener(v -> handleAddInput());
+
 		((AppCompatActivity) getActivity())
-				.getSupportActionBar().setTitle(R.string.category_user_bricks);
+				.getSupportActionBar().setTitle(R.string.brick_add_new_user_brick);
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/fragment/AddUserBrickFragment.java:59]: Null-ptr check, please.

File: catroid/src/main/res/layout/fragment_add_new_user_brick.xml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+            style="@style/BrickText.SingleLine"
+            android:labelFor="@+id/brick_user_brick_edit_text"
+            android:text="@string/brick_user_defined">
+        </TextView>
+        </org.catrobat.catroid.ui.BrickLayout>
+
+    <Button
+        android:id="@+id/add_label"
+        android:layout_width="match_parent"
+        android:layout_height="60dp"
```
Comment (gPathpp) [inline catroid/src/main/res/layout/fragment_add_new_user_brick.xml:51]: Is there any existing style for buttons we could use here?
If so, please use it, else please add one.

File: catroid/src/main/java/org/catrobat/catroid/ui/SpriteActivity.java [hunk 8]
```diff
@@ -260,6 +260,8 @@ public void onBackPressed() {
 		super.onBackPressed();
 	}
 
+
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/SpriteActivity.java:263]: Remove these new blank lines, please.

File: catroid/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1052,6 +1052,13 @@
     <string name="category_assertions">Testing</string>
     <!--  -->
 
+    <!-- User Defined Bricks -->
+    <string name="brick_user_defined">Your brick</string>
```
Comment (wslany) [inline catroid/src/main/res/values/strings.xml:1056]:
```suggestion
```

The blue brick should initially be empty. No text shall be shown there.

File: catroid/src/main/res/layout/fragment_add_new_user_brick.xml [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+        style="@style/BrickContainer.Look.Medium"
+        android:layout_marginTop="10dp"
+        android:background="@drawable/brick_darkblue_1h">
+
+        <include layout="@layout/icon_brick_category_look" />
+
+        <TextView
+            style="@style/BrickText.SingleLine"
+            android:labelFor="@+id/brick_user_brick_edit_text"
+            android:text="@string/brick_user_defined">
```
Comment (wslany) [inline catroid/src/main/res/layout/fragment_add_new_user_brick.xml:46]:
```suggestion
            android:labelFor="@+id/brick_user_brick_edit_text">
```

Or use android:text="", i.e., an empty string if necessary. 
*Reason:* The blue brick should initially be empty. No text shall be shown there.

File: catroid/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -1052,6 +1052,13 @@
...
+    <string name="brick_user_defined_add_label">Add label</string>
+    <string name="brick_user_defined_add_input">Add input</string>
+    <string name="brick_add_new_user_brick">Add new User Brick</string>
```
Comment (wslany) [inline catroid/src/main/res/values/strings.xml:1059]:
```suggestion
    <string name="brick_add_new_user_brick">Add new brick</string>
```

*Reason:* The term "user brick" is only relevant for our developers, but shall not be visible for our users. Also, we try to use normal English capitalization rules.

**Comment by wslany** [comment]
> Please discuss with @wslany about the .9-files and the test.

PLEASE FIX: Re the .9 files: Please fix the static analysis warnings on Jenkins.

SUGGESTION: Re tests: Tests would be nice, and should be easy. The tests could then be extended once more functionality is there (via other tickets), but it might make it a little bit easier to write the more complex tests for the next tickets, if there were tests already for the present ticket. It can, however, also be delayed to later-on tickets.

**Comment by wslany** [comment]
Thanks!

**Pull request merged**